src/code/z_demo.c

```diff
@@ -1,99 +1,1634 @@
+                            csCtx->frames++;
+                        }
+                    } else {
+                        if (cmd->base == 0x33BD) {
+                            func_8019F208();
+                        }
+
+                        if (cmd->textId2 != 0xFFFF) {
+                            func_80151938(globalCtx, cmd->textId2);
+                            if (cmd->type == CS_TEXTBOX_TYPE_3) {
+                                D_801BB160 = CS_TEXTBOX_TYPE_3;
+                                if (cmd->textId1 != 0xFFFF) {
+                                    csCtx->frames++;
+                                }
+                            }
+                        } else {
+                            func_801477B4(globalCtx);
+                            csCtx->frames++;
+                        }
+                    }
+                }
+
+                if (dialogState == 5 && func_80147624(globalCtx) != 0) {
+                    func_80152434(globalCtx, cmd->base);
+                }
+            }
+
+            if ((dialogState == 2) && (D_801BB160 == CS_TEXTBOX_TYPE_3)) {
+                csCtx->frames--;
+                D_801BB124++;
+            }
+
+            if (originalCsFrames == csCtx->frames) {
+                Interface_ChangeAlpha(1);
+                D_801BB124 = 0;
+                D_801BB128 = 0;
+                func_80161C0C();
+            } else {
+                func_80161BE0(1);
+            }
+        }
+    }
+}
+
+void func_800ECD7C(CutsceneContext* csCtx, u8** cutscenePtr, s16 index) {
+    s32 i;
+    s32 count;
+
+    bcopy(*cutscenePtr, &count, sizeof(s32));
+    *cutscenePtr += 4;
+
+    for (i = 0; i < count; i++) {
+        CsCmdActorAction* actorAction = *(CsCmdActorAction**)cutscenePtr;
+
+        if ((csCtx->frames >= actorAction->startFrame) && (csCtx->frames < actorAction->endFrame)) {
+            csCtx->actorActions[index] = actorAction;
+        }
+
+        *cutscenePtr += sizeof(CsCmdActorAction);
+    }
+}
+
+#ifdef NON_MATCHING
+void Cutscene_ProcessCommands(GlobalContext* globalCtx, CutsceneContext* csCtx, u8* cutscenePtr) {
+    s32 i;
+    s16 phi_s0;
+    s32 totalEntries;
+    s32 j;
+    u32 cmdType; // sp5C
+    s32 pad;
+    s32 cmdEntries;       // sp58
+    s32 cutsceneEndFrame; // sp50
+    s16 phi_s0_23;
+    CsCmdBase* cmd;
+
+    bcopy(cutscenePtr, &totalEntries, 4);
+    cutscenePtr += 4;
+    bcopy(cutscenePtr, &cutsceneEndFrame, 4);
+    cutscenePtr += 4;
+
+    if (((u16)cutsceneEndFrame < csCtx->frames) && (globalCtx->sceneLoadFlag != 0x14) && (csCtx->state != CS_STATE_4)) {
+        csCtx->state = CS_STATE_3;
+        return;
+    }
+
+    for (i = 0; i < totalEntries; i++) {
+        bcopy(cutscenePtr, &cmdType, 4);
+        cutscenePtr += 4;
+
+        if (cmdType == 0xFFFFFFFF) {
+            break;
+        }
+
+        if (((cmdType >= 0x64) && (cmdType < 0x96)) || (cmdType == 0xC9) || ((cmdType >= 0x1C2) && (cmdType < 0x258))) {
```

Comment:
Actually, I was thinking about this. If we change this numbers to be dec instead of hex, then we should probably start using dec on functions which use an `actorActionCmd` (usually function calls like `Cutscene_CheckActorAction` and `Cutscene_GetActorActionIndex`)

Reply:
That sounds sensible. You want to do it in this PR?

Reply:
I can try